sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -997,45 +993,51 @@ static void property_callback(
     mqtt_receive_message();
   }
-  else if (az_succeeded(
-               rc = pnp_thermostat_process_property_update(
-                   &hub_client,
-                   &thermostat_1,
-                   component_name,
-                   property_name,
-                   &property_value,
-                   version,
-                   &mqtt_message)))
-  {
-    // Send response to the updated property.
-    mqtt_publish_message(mqtt_message.topic, mqtt_message.out_payload_span, MQTT_PUBLISH_QOS);
-    LOG_SUCCESS("Client sent Temperature Sensor 1 reported property message:");
-    LOG_AZ_SPAN("Payload:", mqtt_message.out_payload_span);
-
-    // Receive the response from the server.
-    mqtt_receive_message();
-  }
-  else if (az_succeeded(
-               rc = pnp_thermostat_process_property_update(
-                   &hub_client,
-                   &thermostat_2,
-                   component_name,
-                   property_name,
-                   &property_value,
-                   version,
-                   &mqtt_message)))
-  {
-    // Send response to the updated property
-    mqtt_publish_message(mqtt_message.topic, mqtt_message.out_payload_span, MQTT_PUBLISH_QOS);
-    LOG_SUCCESS("Client sent Temperature Sensor 2 reported property message.");
-    LOG_AZ_SPAN("Payload:", mqtt_message.out_payload_span);
-
-    // Receive the response from the server.
-    mqtt_receive_message();
-  }
   else
   {
-    LOG_ERROR("Failed to update a property: az_result return code 0x%08x.", rc);
-    exit(rc);
+    rc = pnp_thermostat_process_property_update(
+        &hub_client,
+        &thermostat_1,
+        component_name,
+        property_name,
+        &property_value,
+        version,
+        &mqtt_message);
+    if (az_succeeded(rc))
+    {
+      // Send response to the updated property.
+      mqtt_publish_message(mqtt_message.topic, mqtt_message.out_payload_span, MQTT_PUBLISH_QOS);
+      LOG_SUCCESS("Client sent Temperature Sensor 1 reported property message:");
+      LOG_AZ_SPAN("Payload:", mqtt_message.out_payload_span);
+
+      // Receive the response from the server.
+      mqtt_receive_message();
+    }
+    else
+    {
+      rc = pnp_thermostat_process_property_update(
+          &hub_client,
+          &thermostat_2,
+          component_name,
+          property_name,
+          &property_value,
+          version,
+          &mqtt_message);
+      if (az_succeeded(rc))
+      {
+        // Send response to the updated property
+        mqtt_publish_message(mqtt_message.topic, mqtt_message.out_payload_span, MQTT_PUBLISH_QOS);
+        LOG_SUCCESS("Client sent Temperature Sensor 2 reported property message.");
+        LOG_AZ_SPAN("Payload:", mqtt_message.out_payload_span);
+
+        // Receive the response from the server.
+        mqtt_receive_message();
+      }
+      else
+      {
+        LOG_ERROR("Failed to update a property: az_result return code 0x%08x.", rc);
+        exit(rc);
+      }
```

Comment:
nit: Oh...this particular level of nesting is too deep. @momuno - try to refactor this so it doesn't need several if/else blocks.

Reply:
I think I fixed this in my newest PR.  But just now seeing this comment!